### ISSUE TITLE
Add filters to fetcher 

### ISSUE THREAD

**Issue by abhay-raizada** [PrimalHQ/primal-caching-service#22]
Hello Team,

It'd be great to have an optional feature to fetch events only based on specified filters instead of fetching everything on nostr, it would enable the caching service to be hosted on smaller hosts, and only for usecase specific tasks, IMO it would make it very versatile. 

Are there any plans to do this? I can help contribute to this area as well.

**Comment by pritk** [comment]
hi, filters for fetching are specified here:

https://github.com/PrimalHQ/primal-caching-service/blob/ef7d38933885dc25b0121f725de980c56e78052e/src/fetching.jl#L110-L111

**Comment by abhay-raizada** [comment]
yes, but this is only author right? do you mean i can add new filters here?

**Comment by pritk** [comment]
yes, this is place in code where you can add different filters if you want.
